### ISSUE TITLE
[Linux] FreeSO crashes when changing to neighbourhood view

### ISSUE THREAD

**Issue by rodneyrod** [riperiperi/FreeSO#104]
Steps to reproduce:
1. Follow steps in this forum post to install.
2. Run either of the launchers
3. Update, restart the app, login, then try to change to the neighbourhood view

Output of error:
`exception inside UnhandledException handler: The type initializer for 'System.Console' threw an exception.

[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
  at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x00028] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
  at System.TermInfoReader..ctor (System.String term, System.String filename) [0x0005f] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x00055] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
  at System.ConsoleDriver..cctor () [0x0004d] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x00007] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
  at System.Console..cctor () [0x0008e] in <0cc37f4786fa491387f4cb0ad6d68b47>:0 
   --- End of inner exception stack trace ---
  at FSO.Client.FSOProgram.CurrentDomain_UnhandledException (System.Object sender, System.UnhandledExceptionEventArgs e) [0x0000b] in <40b2c171dc624f2d8a922b51726e2050>:0 
AL lib: (EE) alc_cleanup: 1 device not closed
`

**Comment by neuromancer** [comment]
> Follow steps in this forum post to install.

Which one exactly?

**Comment by rodneyrod** [comment]
@neuromancer Sorry I forgot to post the link. Edited my first post to include it.

**Comment by riperiperi** [comment]
You seem to be missing the actual exception here, which should have been printed before this one I think. The exception you posted is the exception handler (opens a windows forms window) bailing.

If it doesn't print, I'll add a Console.WriteLine as the exception handler on linux platform instead of winforms.

**Comment by rodneyrod** [comment]
@riperiperi Yeah that was all that output on the console for me. I'll test it again with your addition once it gets pushed and post the results.

**Comment by haffmans** [comment]
I'm getting this exception using build master-653 (against a self-hosted server of the same build):

~~~
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at FSO.Client.Rendering.City.CityGeometry.RegenMeshVerts (Microsoft.Xna.Framework.Graphics.GraphicsDevice gd, System.Nullable`1[T] range) [0x0045a] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Client.Rendering.City.Terrain.GenerateCityMesh (Microsoft.Xna.Framework.Graphics.GraphicsDevice gd, System.Nullable`1[T] range) [0x00042] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Client.Rendering.City.Terrain.GenerateAssets () [0x00000] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Client.Rendering.City.Terrain.Draw (Microsoft.Xna.Framework.Graphics.GraphicsDevice gfx) [0x00083] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Common.Rendering.Framework._3DLayer.Draw (Microsoft.Xna.Framework.Graphics.GraphicsDevice device) [0x0001e] in <f6e15d95b6d649108b4e335424a18e33>:0 
  at FSO.Common.Rendering.Framework.GameScreen.Draw (Microsoft.Xna.Framework.GameTime time) [0x000a5] in <f6e15d95b6d649108b4e335424a18e33>:0 
  at FSO.Common.Rendering.Framework.Game.Draw (Microsoft.Xna.Framework.GameTime gameTime) [0x00007] in <f6e15d95b6d649108b4e335424a18e33>:0 
  at Microsoft.Xna.Framework.Game.DoDraw (Microsoft.Xna.Framework.GameTime gameTime) [0x0001c] in <b5e52e31afc14490a85267ae0f4e649c>:0 
  at Microsoft.Xna.Framework.Game.Tick () [0x001f3] in <b5e52e31afc14490a85267ae0f4e649c>:0 
  at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop () [0x00021] in <b5e52e31afc14490a85267ae0f4e649c>:0 
  at Microsoft.Xna.Framework.Game.Run (Microsoft.Xna.Framework.GameRunBehavior runBehavior) [0x00082] in <b5e52e31afc14490a85267ae0f4e649c>:0 
  at FSO.Client.TSOGame.Run () [0x00000] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Client.GameStartProxy.Start (System.Boolean useDX) [0x00011] in <622d21402253448ba5fba434200361b6>:0 
  at FSO.Windows.Program.Main (System.String[] args) [0x00017] in <6d307d4ba050497fa133f5fa293b7afe>:0
~~~
